app.rb

```diff
@@ -0,0 +1,82 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+# Initialize data structures
+
+# Method to list all books
+def list_all_books(books)
+  puts 'List of Books:'
+  books.each do |book|
+    puts "Title: #{book.title}, Author: #{book.author}"
+  end
+end
+
+# Method to list all people
+def list_all_people(people)
+  puts 'List of People:'
+  people.each do |person|
+    puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+  end
+end
+
+# Method to create a person (teacher or student)
+def create_person(people, type, id, age, options = {})
+  name = options[:name] || 'Unknown'
+  parent_permission = options.key?(:parent_permission) ? options[:parent_permission] : true
+  specialization = options[:specialization]
+  classroom = options[:classroom]
+
+  if type == 'teacher'
+    person = Teacher.new(id, age, specialization, { name: name, parent_permission: parent_permission })
```

Comment:
- [ ] When creating a teacher you're passing an object as a parameter in this line, due to this when viewing the list of people the name of the person will show as an object as you can see below.
![image](https://github.com/sulaimonaa/oop-school-library/assets/39240820/b09446be-37b2-4a05-a8dd-dc15c06e45c3)